pkg/propertychecker/azure/checker.go

```diff
@@ -0,0 +1,179 @@
+// can meet the specified property selector requirements.
+type PropertyChecker struct {
+	// vmSizeRecommenderClient is the Azure compute client used to generate VM size recommendations
+	// and validate SKU capacity requirements.
+	vmSizeRecommenderClient compute.AttributeBasedVMSizeRecommenderClient
+}
+
+// NewPropertyChecker creates a new PropertyChecker with the given client.
+// The vmSizeRecommenderClient is used to validate SKU capacity requirements.
+func NewPropertyChecker(vmSizeRecommenderClient compute.AttributeBasedVMSizeRecommenderClient) *PropertyChecker {
+	return &PropertyChecker{
+		vmSizeRecommenderClient: vmSizeRecommenderClient,
+	}
+}
+
+// CheckIfMeetSKUCapacityRequirement validates whether a member cluster can meet the specified
+// SKU capacity requirement. It extracts the required SKU and capacity from the property selector
+// requirement and checks to determine if the cluster's Azure subscription
+// and location can provision the requested VM instances.
+//
+// The cluster must have both Azure location and subscription ID labels configured.
+// Returns true if the SKU capacity requirement can be met, false otherwise.
+func (s *PropertyChecker) CheckIfMeetSKUCapacityRequirement(
+	cluster *clusterv1beta1.MemberCluster,
+	req placementv1beta1.PropertySelectorRequirement,
+) (bool, error) {
+	location, err := labels.ExtractLabelFromMemberCluster(cluster, utils.AzureLocationLabelKey)
+	if err != nil {
+		return false, fmt.Errorf("failed to extract Azure location label from cluster %s: %w", cluster.Name, err)
+	}
+
+	subID, err := labels.ExtractLabelFromMemberCluster(cluster, utils.AzureSubscriptionIDLabelKey)
+	if err != nil {
+		return false, fmt.Errorf("failed to extract Azure subscription ID label from cluster %s: %w", cluster.Name, err)
+	}
+
+	capacity, sku, err := extractCapacityRequirements(req)
+	if err != nil {
+		return false, fmt.Errorf("failed to extract capacity requirement: %w", err)
+	}
+
+	request := &computev1.GenerateAttributeBasedRecommendationsRequest{
+		SubscriptionId: subID,
+		Location:       location,
+		RegularPriorityProfile: &computev1.RegularPriorityProfile{
+			CapacityUnitType: computev1.CapacityUnitType_CAPACITY_UNIT_TYPE_VM_INSTANCE_COUNT,
+			TargetCapacity:   capacity,
+		},
+		ResourceProperties: &computev1.ResourceProperties{
+			VmAttributes: &computev1.VMAttributes{
+				AllowedVmSizes: []string{sku},
+			},
+		},
+		RecommendationProperties: &computev1.RecommendationProperties{
+			RestrictionsFilter: computev1.RecommendationProperties_RESTRICTIONS_FILTER_QUOTA_AND_OFFER_RESTRICTIONS,
+		},
+	}
+
+	respObj, err := s.vmSizeRecommenderClient.GenerateAttributeBasedRecommendations(context.Background(), request)
+	if err != nil {
+		return false, fmt.Errorf("failed to generate VM size recommendations from Azure: %w", err)
+	}
+
+	available := false
+	for _, vm := range respObj.RecommendedVmSizes.RegularVmSizes {
+		if vm.Name == sku {
+			available = true
```

Comment:
is it put for defensive coding? if so, please add a comment to call out.

cause [in the server side](https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=/fleet/pkg/compute/svc/vmsizerecommender_service.go&version=GBmaster&line=225&lineEnd=226&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents), we've already filtered out any SKUs which cannot meet.